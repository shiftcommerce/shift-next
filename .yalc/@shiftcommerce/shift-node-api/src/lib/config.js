class Config {
  constructor () {
    this.config = {}
  }

  set (newConfig) {
    this.config = Object.assign(this.config, newConfig)
    return this.config
  }

  get () {
    return this.config
  }

  reset () {
    this.config = {}
  }
}

module.exports = new Config()