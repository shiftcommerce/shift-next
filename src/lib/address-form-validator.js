export default function (address) {
  const requiredFields = ['line_1', 'zipcode', 'city', 'primary_phone', 'email']

  const requiredFieldsPresent = requiredFields.every(key => address[key])
  const noFormErrorsPresent = (Object.values(address.errors).filter(String).length === 0)

  return requiredFieldsPresent && noFormErrorsPresent
}
