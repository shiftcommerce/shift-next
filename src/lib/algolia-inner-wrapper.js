// Lib
import React from 'react'
import Config from './config'
import InitialPropsDelegator from './initial-props-delegator'
import { Configure } from 'react-instantsearch/dom'

// Components
import { InstantSearch } from './instant-search'

export default function algoliaInnerWrapper (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      const {
        algoliaApiKey,
        algoliaAppId,
        algoliaIndexName
      } = Config.get()
      const { resultsState, onSearchStateChange, searchState, ...otherProps } = this.props

      return (
        <InstantSearch
          appId={algoliaAppId}
          apiKey={algoliaApiKey}
          indexName={algoliaIndexName}
          resultsState={resultsState}
          onSearchStateChange={onSearchStateChange}
          searchState={searchState}
        >
          <Configure {...searchState.configure} />
          <Component {...otherProps} />
        </InstantSearch>
      )
    }
  }
}
