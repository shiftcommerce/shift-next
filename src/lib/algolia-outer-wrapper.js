// Libraries
import React from 'react'
import Router from 'next/router'
import qs from 'qs'

// Lib
import Config from './config'
import InitialPropsDelegator from './initial-props-delegator'
import buildSearchStateForURL from './build-search-state-for-url'
import algoliaInnerWrapper from './algolia-inner-wrapper'

// Components
import { findResultsState } from './instant-search'

const MAX_VARIANTS_PER_PRODUCT = 50

export default function algoliaOuterWrapper (NextWrapper, Page) {
  return class extends InitialPropsDelegator(NextWrapper) {
    constructor (props) {
      super(props)
      this.state = {}
      this.onSearchStateChange = this.onSearchStateChange.bind(this)
    }

    static async getInitialProps (appContext) {
      const { query, req } = appContext
      const isServer = !!req

      // Get initial props from the parent
      const appProps = await Object.getPrototypeOf(this).getInitialProps(appContext)

      let searchState = query

      // remove the random undefined key
      delete searchState.undefined

      if (Page.buildAlgoliaStates) {
        searchState = Page.buildAlgoliaStates(appContext)
      }
      // We should always configure this in the state to prevent a onSearchStateChange first render
      searchState.configure = { ...searchState.configure, hitsPerPage: Config.get().algoliaResultsPerPage, distinct: MAX_VARIANTS_PER_PRODUCT }

      let resultsState = {
        _originalResponse: {
          results: [{}]
        }
      }
      if (isServer && Page.algoliaEnabled && Page.algoliaEnabled()) {
        resultsState = await findResultsState(algoliaInnerWrapper(Page), { searchState, ...appProps })
      }

      return {
        ...appProps,
        resultsState,
        searchState
      }
    }

    updateAfter () {
      // Delegate to custom implementation if the page provides one
      if (Page.updateAfter) {
        return Page.updateAfter()
      }

      // Default implementation
      return 700
    }

    onSearchStateChange (searchState) {
      // Delegate to custom implementation if the page provides one
      if (Page.onSearchStateChange) {
        return Page.onSearchStateChange.call(this, searchState)
      }

      // Default implementation
      clearTimeout(this.debouncedSetState)
      this.debouncedSetState = setTimeout(() => {
        const href = this.searchStateToUrl(searchState)
        Router.push(href, href)
      }, this.updateAfter())
      this.setState({ searchState })
    }

    searchStateToUrl (searchState) {
      // Delegate to custom implementation if the page provides one
      if (Page.searchStateToUrl) {
        return Page.searchStateToUrl.call(this, searchState)
      }

      // Default implementation
      const urlSearchState = buildSearchStateForURL(searchState)
      // Build the query string and append it to search path
      return searchState.query ? `/search?${qs.stringify(urlSearchState)}` : '/search'
    }

    static getDerivedStateFromProps (newProps, prevState) {
      if (Page.algoliaGetDerivedStateFromProps) {
        return Page.algoliaGetDerivedStateFromProps.call(this, newProps, prevState)
      }
      return null
    }

    componentDidUpdate (prevProps, prevState) {
      return Page.algoliaComponentDidUpdate && Page.algoliaComponentDidUpdate.call(this, prevProps, prevState)
    }

    componentDidMount () {
      return Page.algoliaComponentDidMount && Page.algoliaComponentDidMount.call(this)
    }

    render () {
      const { resultsState, searchState, ...appProps } = this.props

      return (
        <NextWrapper
          resultsState={resultsState}
          onSearchStateChange={this.onSearchStateChange}
          searchState={
            this.state && this.state.searchState
              ? this.state.searchState
              : searchState
          }
          {...appProps}
        />
      )
    }
  }
}
