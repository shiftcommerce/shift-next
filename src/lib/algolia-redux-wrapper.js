// Lib
import algoliaOuterWrapper from './algolia-outer-wrapper'
import withReduxStore from './with-redux-store'
import withProvider from './with-provider'
import algoliaInnerWrapper from './algolia-inner-wrapper'
import withLayout from './with-layout'

export function algoliaReduxWrapper (ConnectedComponent, Page) {
  return algoliaOuterWrapper(
    withReduxStore(
      withProvider(
        algoliaInnerWrapper(
          withLayout(
            ConnectedComponent
          )
        )
      )
    ),
    Page
  )
}

export function reduxWrapper (ConnectedComponent, Page) {
  return withReduxStore(
    withProvider(
      withLayout(
        ConnectedComponent
      )
    )
  )
}
