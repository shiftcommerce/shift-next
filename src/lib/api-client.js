// Libraries
import qs from 'qs'
import axios from 'axios'

import Config from '../lib/config'

class ApiClient {
  constructor (options = {}) {
    this.client = axios.create({
      baseURL: Config.get().apiHostProxy
    })
  }

  async read (endpoint, queryObject = {}, options = {}) {
    const formattedEndpoint = this.encodeParams(endpoint, queryObject)

    try {
      const response = await this.client.get(formattedEndpoint)
      return { status: response.status, data: response.data }
    } catch (error) {
      console.error('API CLIENT: Error while fetching data', error)
      return { status: error.response.status, data: error.response.data }
    }
  }

  async post (endpoint, body = {}, options = {}) {
    try {
      const response = await this.client.post(endpoint, body)
      return { status: response.status, data: response.data }
    } catch (error) {
      console.error('Error while posting data', error)
      return { status: error.response.status, data: error.response.data }
    }
  }

  async delete (endpoint, options = {}) {
    try {
      const response = await this.client.delete(endpoint)
      return { status: response.status, data: response.data }
    } catch (error) {
      console.error('Error during delete request', error)
      return { status: error.response.status, data: error.response.data }
    }
  }

  encodeParams (endpoint, queryObject = {}) {
    if (Object.keys(queryObject).length > 0) {
      const query = qs.stringify(queryObject)
      endpoint = `${endpoint}?${query}`
    }
    return endpoint
  }
}

export default ApiClient
