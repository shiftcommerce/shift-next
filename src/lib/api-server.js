const axios = require('axios')
const qs = require('qs')

// libs
const { setCacheHeaders } = require('./set-cache-headers')

const auth = {
  username: process.env.API_TENANT,
  password: process.env.API_ACCESS_TOKEN
}

const headers = {
  'Content-Type': 'application/vnd.api+json',
  'Accept': 'application/vnd.api+json'
}

const fetchData = async (queryObject, url) => {
  const query = qs.stringify(queryObject)

  try {
    const response = await axios.get(`${process.env.API_HOST}/${url}?${query}`, {
      auth: auth,
      headers: headers
    })

    setCacheHeaders(response)
    return response
  } catch (error) {
    console.error(error)
    return error.response
  }
}

module.exports = { fetchData }
