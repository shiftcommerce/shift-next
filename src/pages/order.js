//  Libraries
import React, { Component } from 'react'
import Link from 'next/link'
import t from 'typy'

// Objects
import { Button, Image, OrderSummary } from '@shiftcommerce/shift-react-components'

class OrderPage extends Component {
  renderHeader (order) {
    const customerName = t(order, 'shipping_address.attributes.first_name').safeObject

    return (
      <div className='c-order__header'>
        <div className='c-order__header-tick'>âœ”</div>
        <div className='c-order__header-text'>
          <p className='c-order__header-text-id u-text-color--orange u-bold'>Order #{ order.id }</p>
          <h3 className='u-bold'>Thank you { customerName }!</h3>
        </div>
      </div>
    )
  }

  renderOrderConfirmationMessage (order) {
    return (
      <div className='c-order__detail-confirmation-message'>
        <h2>Your order is confirmed</h2>
        <p>We've accepted your order and we're getting it ready. A confirmation email has been sent to <span className='u-text-color--orange'>{ order.email }</span>. Come back to this page for updates on your order status.</p>
      </div>
    )
  }

  renderShippingAddress (order) {
    const shippingAddress = t(order, 'shipping_address.attributes').safeObject

    return (
      <div className='c-order__detail-shipping-address'>
        <h2>Shipping Address</h2>
        <p>{ shippingAddress.first_name } { shippingAddress.last_name }</p>
        <p>{ shippingAddress.address_line_1 }, { shippingAddress.address_line_2 }, { shippingAddress.postcode }</p>
      </div>
    )
  }

  renderShippingMethod (order) {
    const shippingMethod = t(order, 'shipping_method.attributes').safeObject

    return (
      <div className='c-order__detail-shipping-method'>
        <h2>Shipping Method</h2>
        <p>{ shippingMethod.label }</p>
        <p>Estimated Delivery:<span> { shippingMethod.description }</span></p>
      </div>
    )
  }

  renderCardImage (cardDetails) {
    if (cardDetails.brand === 'Visa') {
      return (
        <Image src='/static/payments/visa.svg' className='c-order__detail-payment-method-card-image' />
      )
    }
  }

  renderPaymentMethod (order) {
    const billingAddress = t(order, 'billing_address.attributes').safeObject
    const cardDetails = t(order, 'cardToken.card').safeObject
    const displayMonth = { 1: '01', 2: '02', 3: '02', 4: '04', 5: '05', 6: '06', 7: '07', 8: '08', 9: '09', 10: '10', 11: '11', 12: '12' }

    return (
      <div className='c-order__detail-payment-method'>
        <h2>Payment Method</h2>
        <div className='c-order__detail-payment-method-card'>
          { this.renderCardImage(cardDetails) }
          <div className='c-order__detail-payment-method-card-details'>
            <span className='u-bold'>**** **** **** { cardDetails.last4 }</span>
            <span className='u-bold'>Exp: { displayMonth[cardDetails.exp_month] }/{ cardDetails.exp_year }</span>
          </div>
        </div>
        <p className='u-bold'>Billing Address:</p>
        <p>{ billingAddress.first_name } { billingAddress.last_name }</p>
        <p>{ billingAddress.address_line_1 }, { billingAddress.address_line_2 }, { billingAddress.postcode }</p>
      </div>
    )
  }

  renderContinueShoppingButton () {
    return (
      <div className='c-order__button'>
        <Link href='/'>
          <Button
            aria-label='Continue Shopping'
            label='Continue shopping'
            status='positive'
            className='c-order__button--continue o-button--sml'
            type='button'
          />
        </Link>
      </div>
    )
  }

  renderOrderDetails (order) {
    return (
      <div className='c-order__detail'>
        { this.renderOrderConfirmationMessage(order) }
        { this.renderShippingAddress(order) }
        { this.renderShippingMethod(order) }
        { this.renderPaymentMethod(order) }
      </div>
    )
  }

  render () {
    const { order } = this.props

    if (order.loading) {
      return (
        <p>Loading...</p>
      )
    } else if (order.error) {
      return (
        <p>{ order.error }</p>
      )
    } else {
      return (
        <div className='c-order'>
          { this.renderHeader(order) }
          <div className='c-order__body'>
            <div className='c-order__body-panel-left'>
              { this.renderOrderDetails(order) }
            </div>
            <div className='c-order__body-panel-right'>
              <OrderSummary order={order} />
              { this.renderContinueShoppingButton() }
            </div>
          </div>
        </div>
      )
    }
  }
}

export default OrderPage
