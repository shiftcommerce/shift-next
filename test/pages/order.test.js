// Libraries
import { Provider } from 'react-redux'
import { createMockStore } from 'redux-test-utils'

// Pages
import OrderPage from '../../src/pages/order'

// Lib
import { decimalPrice } from '../../src/lib/decimal-price'

// Fixtures
import order from '../fixtures/confirmation-order'

jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {}
}))

describe('Page Rendering:', () => {
  test('renders correctly', () => {
    // Arrange
    const initialState = { }
    const lineItems = order.line_items
    const shippingAddress = order.shipping_address.attributes
    const shippingMethod = order.shipping_method.attributes
    const billingAddress = order.billing_address.attributes
    const cardDetails = order.cardToken.card

    // Act
    const wrapper = mount(
      <Provider store={createMockStore(initialState)}>
        <OrderPage order={order} />
      </Provider>
    )

    // Assert
    expect(wrapper).toMatchSnapshot()

    // Verify order summary info
    expect(wrapper).toIncludeText(lineItems[0].title)
    expect(wrapper).toIncludeText(decimalPrice(order.total_inc_tax))

    // Verify shipping address info
    expect(wrapper).toIncludeText(shippingAddress.first_name)
    expect(wrapper).toIncludeText(shippingAddress.last_name)
    expect(wrapper).toIncludeText(shippingAddress.address_line_1)
    expect(wrapper).toIncludeText(shippingAddress.postcode)

    // Verify shipping method info
    expect(wrapper).toIncludeText(shippingMethod.label)
    expect(wrapper).toIncludeText(shippingMethod.description)

    // Verify billing address info
    expect(wrapper).toIncludeText(billingAddress.first_name)
    expect(wrapper).toIncludeText(billingAddress.last_name)
    expect(wrapper).toIncludeText(billingAddress.address_line_1)
    expect(wrapper).toIncludeText(billingAddress.postcode)

    // Verify card details info
    expect(wrapper).toIncludeText(cardDetails.last4)
    expect(wrapper).toIncludeText(cardDetails.exp_month)
    expect(wrapper).toIncludeText(cardDetails.exp_year)
  })
})
