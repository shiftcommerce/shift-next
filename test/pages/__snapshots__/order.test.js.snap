// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Rendering: renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <OrderPage
    order={
      Object {
        "billing_address": Object {
          "attributes": Object {
            "address_line_1": "104 Quay One",
            "address_line_2": "",
            "city": "Leeds",
            "country": "GB",
            "first_name": "john",
            "last_name": "smith",
            "postcode": "LS9 8DS",
          },
          "id": "",
          "type": "addresses",
        },
        "cardToken": Object {
          "card": Object {
            "address_city": "Leeds",
            "address_country": "GB",
            "address_line1": "104 Quay One",
            "address_line1_check": "unchecked",
            "address_line2": "",
            "address_state": "Some state",
            "address_zip": "LS9 8DS",
            "address_zip_check": "unchecked",
            "brand": "Visa",
            "country": "US",
            "cvc_check": "unchecked",
            "dynamic_last4": null,
            "exp_month": 9,
            "exp_year": 2029,
            "funding": "credit",
            "id": "card_1DRhVLIZ9svPgtYNHTG1Vare",
            "last4": "4242",
            "metadata": Object {},
            "name": "john smith",
            "object": "card",
            "tokenization_method": null,
          },
          "client_ip": "90.198.102.132",
          "created": 1541083839,
          "id": "tok_1DRhVLIZ9svPgtYNDsRIAYyp",
          "livemode": false,
          "object": "token",
          "type": "card",
          "used": false,
        },
        "cardTokenRequested": false,
        "card_errors": false,
        "channel": "web",
        "coupons_applied": Array [],
        "currency": "GBP",
        "customer_account_id": null,
        "deallocated": false,
        "deallocated_at": null,
        "discount_summaries": Array [
          Object {
            "attributes": Object {
              "name": "",
              "promotion_id": 0,
              "total": 0,
            },
            "id": "",
            "type": "discount_summaries",
          },
        ],
        "email": "john.smith@shiftcommerce.com",
        "error": false,
        "free_shipping": false,
        "id": "95",
        "ip_address": "90.198.102.132",
        "line_items": Array [
          Object {
            "gift": false,
            "gift_unit_quantity": 0,
            "id": "127",
            "line_item_discounts": Array [],
            "sku": "1425689355217",
            "taxes": 0,
            "title": "SCS-04-CJ coffee jug 600ml",
            "tracking_url": null,
            "unit_price": 90.31,
            "unit_quantity": 1,
          },
          Object {
            "gift": false,
            "gift_unit_quantity": 0,
            "id": "128",
            "line_item_discounts": Array [],
            "sku": "bdm1",
            "taxes": 0,
            "title": "Brazil Daterra Masterpiece - Curupira",
            "tracking_url": null,
            "unit_price": 16.5,
            "unit_quantity": 1,
          },
        ],
        "loading": false,
        "paymentError": null,
        "placed_at": "2018-11-01T14:50:39.968Z",
        "reference": "_95_",
        "shipping_address": Object {
          "attributes": Object {
            "address_line_1": "228 Query Avenue",
            "address_line_2": "",
            "city": "Manchester",
            "country": "GB",
            "first_name": "Terry",
            "last_name": "Tibbs",
            "postcode": "M1 1TP",
          },
          "id": "",
          "type": "addresses",
        },
        "shipping_discount": 0,
        "shipping_discount_name": null,
        "shipping_method": Object {
          "attributes": Object {
            "created_at": "2018-11-01T14:50:35.544Z",
            "description": "Sunday 2nd December",
            "label": "Super Saver",
            "meta_attributes": Object {},
            "reference": "super_saver",
            "sku": "super_saver",
            "sub_total": 3.45,
            "tax": 2,
            "tax_rate": 20,
            "total": 3.45,
            "updated_at": "2018-11-01T14:50:35.544Z",
          },
          "id": "1",
          "type": "shipping_methods",
        },
        "shipping_total": 3.45,
        "status": "pending",
        "sub_total": 106.81,
        "tax": 0,
        "test": true,
        "total": 110.26,
      }
    }
  >
    <div
      className="c-order"
    >
      <div
        className="c-order__header"
      >
        <div
          className="c-order__header-tick"
        >
          âœ”
        </div>
        <div
          className="c-order__header-text"
        >
          <p
            className="c-order__header-text-id u-text-color--orange u-bold"
          >
            Order #
            95
          </p>
          <h3
            className="u-bold"
          >
            Thank you 
            Terry
            !
          </h3>
        </div>
      </div>
      <div
        className="c-order__body"
      >
        <div
          className="c-order__body-panel-left"
        >
          <div
            className="c-order__detail"
          >
            <div
              className="c-order__detail-confirmation-message"
            >
              <h2>
                Your order is confirmed
              </h2>
              <p>
                We've accepted your order and we're getting it ready. A confirmation email has been sent to 
                <span
                  className="u-text-color--orange"
                >
                  john.smith@shiftcommerce.com
                </span>
                . Come back to this page for updates on your order status.
              </p>
            </div>
            <div
              className="c-order__detail-shipping-address"
            >
              <h2>
                Shipping Address
              </h2>
              <p>
                Terry
                 
                Tibbs
              </p>
              <p>
                228 Query Avenue
                , 
                , 
                M1 1TP
              </p>
            </div>
            <div
              className="c-order__detail-shipping-method"
            >
              <h2>
                Shipping Method
              </h2>
              <p>
                Super Saver
              </p>
              <p>
                Estimated Delivery:
                <span>
                   
                  Sunday 2nd December
                </span>
              </p>
            </div>
            <div
              className="c-order__detail-payment-method"
            >
              <h2>
                Payment Method
              </h2>
              <div
                className="c-order__detail-payment-method-card"
              >
                <Image
                  className="c-order__detail-payment-method-card-image"
                  src="/static/payments/visa.svg"
                >
                  <div
                    className="c-order__detail-payment-method-card-image"
                  >
                    <picture>
                      <img
                        className="o-image"
                        src="/static/payments/visa.svg"
                        srcSet="/static/payments/visa.svg 769w"
                      />
                    </picture>
                  </div>
                </Image>
                <div
                  className="c-order__detail-payment-method-card-details"
                >
                  <span
                    className="u-bold"
                  >
                    **** **** **** 
                    4242
                  </span>
                  <span
                    className="u-bold"
                  >
                    Exp: 
                    09
                    /
                    2029
                  </span>
                </div>
              </div>
              <p
                className="u-bold"
              >
                Billing Address:
              </p>
              <p>
                john
                 
                smith
              </p>
              <p>
                104 Quay One
                , 
                , 
                LS9 8DS
              </p>
            </div>
          </div>
        </div>
        <div
          className="c-order__body-panel-right"
        >
          <OrderSummary
            order={
              Object {
                "billing_address": Object {
                  "attributes": Object {
                    "address_line_1": "104 Quay One",
                    "address_line_2": "",
                    "city": "Leeds",
                    "country": "GB",
                    "first_name": "john",
                    "last_name": "smith",
                    "postcode": "LS9 8DS",
                  },
                  "id": "",
                  "type": "addresses",
                },
                "cardToken": Object {
                  "card": Object {
                    "address_city": "Leeds",
                    "address_country": "GB",
                    "address_line1": "104 Quay One",
                    "address_line1_check": "unchecked",
                    "address_line2": "",
                    "address_state": "Some state",
                    "address_zip": "LS9 8DS",
                    "address_zip_check": "unchecked",
                    "brand": "Visa",
                    "country": "US",
                    "cvc_check": "unchecked",
                    "dynamic_last4": null,
                    "exp_month": 9,
                    "exp_year": 2029,
                    "funding": "credit",
                    "id": "card_1DRhVLIZ9svPgtYNHTG1Vare",
                    "last4": "4242",
                    "metadata": Object {},
                    "name": "john smith",
                    "object": "card",
                    "tokenization_method": null,
                  },
                  "client_ip": "90.198.102.132",
                  "created": 1541083839,
                  "id": "tok_1DRhVLIZ9svPgtYNDsRIAYyp",
                  "livemode": false,
                  "object": "token",
                  "type": "card",
                  "used": false,
                },
                "cardTokenRequested": false,
                "card_errors": false,
                "channel": "web",
                "coupons_applied": Array [],
                "currency": "GBP",
                "customer_account_id": null,
                "deallocated": false,
                "deallocated_at": null,
                "discount_summaries": Array [
                  Object {
                    "attributes": Object {
                      "name": "",
                      "promotion_id": 0,
                      "total": 0,
                    },
                    "id": "",
                    "type": "discount_summaries",
                  },
                ],
                "email": "john.smith@shiftcommerce.com",
                "error": false,
                "free_shipping": false,
                "id": "95",
                "ip_address": "90.198.102.132",
                "line_items": Array [
                  Object {
                    "gift": false,
                    "gift_unit_quantity": 0,
                    "id": "127",
                    "line_item_discounts": Array [],
                    "sku": "1425689355217",
                    "taxes": 0,
                    "title": "SCS-04-CJ coffee jug 600ml",
                    "tracking_url": null,
                    "unit_price": 90.31,
                    "unit_quantity": 1,
                  },
                  Object {
                    "gift": false,
                    "gift_unit_quantity": 0,
                    "id": "128",
                    "line_item_discounts": Array [],
                    "sku": "bdm1",
                    "taxes": 0,
                    "title": "Brazil Daterra Masterpiece - Curupira",
                    "tracking_url": null,
                    "unit_price": 16.5,
                    "unit_quantity": 1,
                  },
                ],
                "loading": false,
                "paymentError": null,
                "placed_at": "2018-11-01T14:50:39.968Z",
                "reference": "_95_",
                "shipping_address": Object {
                  "attributes": Object {
                    "address_line_1": "228 Query Avenue",
                    "address_line_2": "",
                    "city": "Manchester",
                    "country": "GB",
                    "first_name": "Terry",
                    "last_name": "Tibbs",
                    "postcode": "M1 1TP",
                  },
                  "id": "",
                  "type": "addresses",
                },
                "shipping_discount": 0,
                "shipping_discount_name": null,
                "shipping_method": Object {
                  "attributes": Object {
                    "created_at": "2018-11-01T14:50:35.544Z",
                    "description": "Sunday 2nd December",
                    "label": "Super Saver",
                    "meta_attributes": Object {},
                    "reference": "super_saver",
                    "sku": "super_saver",
                    "sub_total": 3.45,
                    "tax": 2,
                    "tax_rate": 20,
                    "total": 3.45,
                    "updated_at": "2018-11-01T14:50:35.544Z",
                  },
                  "id": "1",
                  "type": "shipping_methods",
                },
                "shipping_total": 3.45,
                "status": "pending",
                "sub_total": 106.81,
                "tax": 0,
                "test": true,
                "total": 110.26,
              }
            }
          >
            <div
              className="c-order__summary"
            >
              <div
                className="c-order__summary-header"
              >
                <div
                  className="c-order__summary-header-title"
                >
                  <h2>
                    Your Order
                  </h2>
                </div>
                <div
                  className="c-order__summary-header-total"
                >
                  <h2>
                    Â£
                    110.26
                  </h2>
                </div>
              </div>
              <div
                className="c-order__summary-line-item"
                key="1425689355217"
              >
                <div
                  className="c-order__summary-line-item-image"
                >
                  <Image
                    alt="SCS-04-CJ coffee jug 600ml"
                    aria-label="SCS-04-CJ coffee jug 600ml"
                    src="/static/placeholder.png"
                  >
                    <div>
                      <picture>
                        <img
                          alt="SCS-04-CJ coffee jug 600ml"
                          aria-label="SCS-04-CJ coffee jug 600ml"
                          className="o-image"
                          src="/static/placeholder.png"
                          srcSet="/static/placeholder.png 769w"
                        />
                      </picture>
                    </div>
                  </Image>
                </div>
                <div
                  className="c-order__summary-line-item-details"
                >
                  <p
                    className="c-order__summary-line-item-details-title u-bold"
                  >
                    SCS-04-CJ coffee jug 600ml
                  </p>
                  <p
                    className="c-order__summary-line-item-details-sku"
                  >
                    1425689355217
                  </p>
                  <div
                    className="c-order__summary-line-item-details-param"
                    key="colour"
                  >
                    <div
                      className="c-order__summary-line-item-details-param-key"
                    >
                      <p>
                        colour
                        :
                      </p>
                    </div>
                    <div
                      className="c-order__summary-line-item-details-param-value"
                    >
                      <p>
                        Grey
                      </p>
                    </div>
                  </div>
                  <div
                    className="c-order__summary-line-item-details-param"
                    key="size"
                  >
                    <div
                      className="c-order__summary-line-item-details-param-key"
                    >
                      <p>
                        size
                        :
                      </p>
                    </div>
                    <div
                      className="c-order__summary-line-item-details-param-value"
                    >
                      <p>
                        L
                      </p>
                    </div>
                  </div>
                  <div
                    className="c-order__summary-line-item-details-param"
                    key="quantity"
                  >
                    <div
                      className="c-order__summary-line-item-details-param-key"
                    >
                      <p>
                        quantity
                        :
                      </p>
                    </div>
                    <div
                      className="c-order__summary-line-item-details-param-value"
                    >
                      <p>
                        1
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="c-order__summary-line-item"
                key="bdm1"
              >
                <div
                  className="c-order__summary-line-item-image"
                >
                  <Image
                    alt="Brazil Daterra Masterpiece - Curupira"
                    aria-label="Brazil Daterra Masterpiece - Curupira"
                    src="/static/placeholder.png"
                  >
                    <div>
                      <picture>
                        <img
                          alt="Brazil Daterra Masterpiece - Curupira"
                          aria-label="Brazil Daterra Masterpiece - Curupira"
                          className="o-image"
                          src="/static/placeholder.png"
                          srcSet="/static/placeholder.png 769w"
                        />
                      </picture>
                    </div>
                  </Image>
                </div>
                <div
                  className="c-order__summary-line-item-details"
                >
                  <p
                    className="c-order__summary-line-item-details-title u-bold"
                  >
                    Brazil Daterra Masterpiece - Curupira
                  </p>
                  <p
                    className="c-order__summary-line-item-details-sku"
                  >
                    bdm1
                  </p>
                  <div
                    className="c-order__summary-line-item-details-param"
                    key="colour"
                  >
                    <div
                      className="c-order__summary-line-item-details-param-key"
                    >
                      <p>
                        colour
                        :
                      </p>
                    </div>
                    <div
                      className="c-order__summary-line-item-details-param-value"
                    >
                      <p>
                        Grey
                      </p>
                    </div>
                  </div>
                  <div
                    className="c-order__summary-line-item-details-param"
                    key="size"
                  >
                    <div
                      className="c-order__summary-line-item-details-param-key"
                    >
                      <p>
                        size
                        :
                      </p>
                    </div>
                    <div
                      className="c-order__summary-line-item-details-param-value"
                    >
                      <p>
                        L
                      </p>
                    </div>
                  </div>
                  <div
                    className="c-order__summary-line-item-details-param"
                    key="quantity"
                  >
                    <div
                      className="c-order__summary-line-item-details-param-key"
                    >
                      <p>
                        quantity
                        :
                      </p>
                    </div>
                    <div
                      className="c-order__summary-line-item-details-param-value"
                    >
                      <p>
                        1
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="c-order__summary-totals"
              >
                <Sticker>
                  <div
                    aria-label="Order total summary"
                    className="u-sticky"
                  >
                    <div>
                      <dl
                        aria-label="Order subtotal"
                      >
                        <dt>
                          Total products:
                        </dt>
                        <dd>
                          Â£
                          106.81
                        </dd>
                      </dl>
                      <dl
                        aria-label="Order shipping costs"
                      >
                        <dt>
                          Shipping costs:
                        </dt>
                        <dd>
                          Â£
                          3.45
                        </dd>
                      </dl>
                      <dl
                        aria-label="Order total"
                        className="u-bold"
                      >
                        <dt>
                          TOTAL:
                        </dt>
                        <dd>
                          Â£
                          110.26
                        </dd>
                      </dl>
                      <dl>
                        <dt>
                          * Including VAT
                        </dt>
                      </dl>
                    </div>
                  </div>
                </Sticker>
              </div>
            </div>
          </OrderSummary>
          <div
            className="c-order__button"
          >
            <Link
              href="/"
            >
              <Button
                aria-label="Continue Shopping"
                className="c-order__button--continue o-button--sml"
                label="Continue shopping"
                onClick={[Function]}
                status="positive"
                type="button"
              >
                <button
                  aria-label="Continue Shopping"
                  className="c-order__button--continue o-button--sml o-button o-button--positive"
                  disabled={false}
                  onClick={[Function]}
                  role="button"
                  tabIndex="0"
                  type="button"
                >
                  Continue shopping
                </button>
              </Button>
            </Link>
          </div>
        </div>
      </div>
    </div>
  </OrderPage>
</Provider>
`;
